### PR TITLE
Rover: limit all turn rates to ATC_TURN_MAX_G

### PR DESCRIPTION
**Pull request by IamPete1**
This moves TURN_MAX_G to ATC_TURN_MAX_G and uses it to limit all turn rates. The g limit is constrained to a min of 0.1G and it will not constrain turn rate to lower than 1 deg/sec

Alternate to: https://github.com/ArduPilot/ardupilot/pull/16333 from @srmainwaring 

Tested with this lua script. 
```
function update()
  local accel = ahrs:get_accel()
  accel:z(0)

  gcs:send_named_float('G',accel:length()/9.81)

  return update, 10
end

return update()
```

### CODE REVIEW

**Comment by IamPete1** [comment]
To elaborate a little more, the exiting G limit works by trying to slow down not by altering the turn rate. That code is unchanged, so the vehicle will still slow down. This then constrains the turn rate so it cant exceed that limit, so this should only kick in if your going faster than than desired speed.

@srmainwaring hit this on his land yacht because we don't do speed control on sailboats, its either going or not, this meant the existing method of reducing the desired speed does not work.

File: libraries/APM_Control/AR_AttitudeControl.h [hunk 1]
```diff
@@ -86,6 +86,9 @@ class AR_AttitudeControl {
     // calculate the turn rate in rad/sec given a lateral acceleration (in m/s/s) and speed (in m/s)
     float get_turn_rate_from_lat_accel(float lat_accel, float speed) const;
 
+    // get the g limit lateral acceleration ( m/s/s), returning at least 0.1g
+    float get_g_limit_lateral_accel() { return MAX(_steer_lateral_g_max * GRAVITY_MSS, 0.1f); }
```
Comment (rmackay9) [inline libraries/APM_Control/AR_AttitudeControl.h:90]: this can be const..

File: libraries/APM_Control/AR_AttitudeControl.h [hunk 2]
```diff
@@ -167,6 +170,7 @@ class AR_AttitudeControl {
     AP_Float _stop_speed;           // speed control stop speed.  Motor outputs to zero once vehicle speed falls below this value
     AP_Float _steer_accel_max;      // steering angle acceleration max in deg/s/s
     AP_Float _steer_rate_max;       // steering rate control maximum rate in deg/s
+    AP_Float _steer_lateral_g_max;  // sterring maximum lateral acceleration limit in 'g'
```
Comment (rmackay9) [inline libraries/APM_Control/AR_AttitudeControl.h:173]: "g" is for grams I think.. probably a capital G.. sorry to be nitpicky..

File: libraries/AR_WPNav/AR_WPNav.cpp [hunk 3]
```diff
@@ -216,7 +216,7 @@ bool AR_WPNav::set_desired_location(const struct Location& destination, float ne
         } else {
             // calculate maximum speed that keeps overshoot within bounds
             const float radius_m = fabsf(_overshoot / (cosf(radians(turn_angle_cd * 0.01f)) - 1.0f));
-            _desired_speed_final = MIN(_desired_speed, safe_sqrt(_turn_max_mss * radius_m));
+            _desired_speed_final = MIN(_desired_speed, safe_sqrt(_atc.get_g_limit_lateral_accel() * radius_m));
```
Comment (rmackay9) [inline libraries/AR_WPNav/AR_WPNav.cpp:219]: as mentioned on the call let's try and keep the parameter name, the variable and the accessor all using consistent names.

**Comment by rmackay9** [comment]
Re the discussion of m/s/s vs G you can go ahead keeping it G for now because there's disagreement in the dev team and keeping it consistent with what we have now seems like a safe choice.

File: libraries/APM_Control/AR_AttitudeControl.h (hunk 1 above)
Comment (IamPete1) [reply libraries/APM_Control/AR_AttitudeControl.h]: done, thanks

File: libraries/APM_Control/AR_AttitudeControl.h (hunk 2 above)
Comment (IamPete1) [reply libraries/APM_Control/AR_AttitudeControl.h]: changed to `_turn_lateral_G_max` to better match the param

File: libraries/AR_WPNav/AR_WPNav.cpp (hunk 3 above)
Comment (IamPete1) [reply libraries/AR_WPNav/AR_WPNav.cpp]: changed to `get_turn_lat_accel_max`

**Comment by IamPete1** [comment]
Rebased, re-tested and done some renaming to make more consistent.

**Comment by rmackay9** [comment]
Merged, thanks!

**Comment by rmackay9** [comment]
I suspect this PR is causing this issue https://github.com/ArduPilot/ardupilot/issues/17417 and also is leading to skid-steering vehicles getting stuck at corners.  The issue can be reproduced in SITL with a regular square or triangular mission using a skid-steering vehicle.  I think you'll find it regularly gets stuck for a long time at the corners.

**Pull request merged**